### ISSUE TITLE
User is not redirected back to the login-related KB article when clicking the "Cancel" form button

### ISSUE THREAD

**Issue by emilghittasv** [mozilla/sumo#1531]
**Describe the bug**
As described inside the figma file, "Clicking on the “Cancel” Secondary Button navigates the user back to the page they came from:
If they came from a login-related KB article, the user should get taken back to the article page."

"Alternative: If neither of the above is possible, the user should get taken back to the Firefox Accounts Support landing page."

**Steps to reproduce**
1. Access the https://support.allizom.org/en-US/kb/firefox-accounts-supportmozillaorg article.
2. Click on the "Contact Support" CTA button.
3. Click on the "Cancel" button from the "Contact Support" form.

**Expected behavior**
The user is redirected back to the kb article from step 2.

**Actual behavior**
The user is redirected to the Mozilla Account solutions page -> https://support.allizom.org/en-US/questions/new/mozilla-account

**Screencasts**
![kbcancelredirect](https://github.com/mozilla/sumo/assets/28087895/ff55f4aa-9927-4186-8d61-b252812941f0)

**Desktop:**
 - OS: Windows 10 
 - Browser: Firefox & Chrome.

**Additional context**
This occurs when reaching both the loginless and the normal (logged in) Mozilla Account contact support form

**Comment by emilghittasv** [comment]
@smithellis This also occurs if reaching the form via the "Contact Support" CTA button from the Featured Topics List page -> https://support.allizom.org/en-US/products/mozilla-account/troubleshoot-accounts

The user is redirected to the Mozilla Account Solutions page when clicking the "Cancel" button from the Contact Support form.

**Comment by emilghittasv** [comment]
During the verification of this issue I've noticed a regression affecting the Firefox & Firefox Focus products. When clicking on the form "Cancel" button, the user is redirected to a 404 page -> https://github.com/mozilla/sumo/issues/1551

The issue in this ticket description is now verified fixed in `stage`. 

If the loginless form was accessed via the CTA of a special KB article and the "Cancel" button was clicked inside the form, the user is redirected back to the KB article.

During the verification of this issue I've also noticed that now we have a deviation from the figma doc in which the user is redirected back to the login interstitial page after clicking on the "Cancel" button for the loginless form. Filled under -> https://github.com/mozilla/sumo/issues/1552
